app/controllers/application_controller/ci_processing.rb

```diff
@@ -1609,42 +1607,55 @@ def scanclusters
+                    :task => task_name,
+                    :message => bang.message 
+                  }, :error)
+      end
+    end
+  end
+
   # Common Host button handler routines
   def process_hosts(hosts, task, display_name = nil)
-    hosts, hosts_out_region = filter_ids_in_region(hosts, "Host")
+    hosts, _hosts_out_region = filter_ids_in_region(hosts, "Host")
     return if hosts.empty?
+    task_name = (display_name || task)
 
-    if task == "refresh_ems"
+    case task
+    when "refresh_ems"
       Host.refresh_ems(hosts)
-      #      add_flash("'" + task.titleize + "' initiated for " + pluralize(hosts.length,"Host"))
       add_flash(_("%{task} initiated for %{count_model} from the CFME Database") % {:task => (display_name || Dictionary.gettext(task, :type => :task).titleize), :count_model => pluralize(hosts.length, "Host")})
       AuditEvent.success(:userid => session[:userid], :event => "host_#{task}",
-          :message => "'#{display_name || task}' successfully initiated for #{pluralize(hosts.length, "Host")}",
+          :message => "'#{task_name}' successfully initiated for #{pluralize(hosts.length, "Host")}",
           :target_class => "Host")
-    elsif task == "destroy"
-      Host.where(:id => hosts).order("lower(name)").each do |host|
-        id = host.id
-        host_name = host.name
-        audit = {:event => "host_record_delete_initiated", :message => "[#{host_name}] Record delete initiated", :target_id => id, :target_class => "Host", :userid => session[:userid]}
+    when "destroy"
+      each_host(hosts, task_name) do |host|
+        audit = {:event => "host_record_delete_initiated", :message => "[#{host.name}] Record delete initiated", :target_id => id, :target_class => "Host", :userid => session[:userid]}
         AuditEvent.success(audit)
       end
       Host.destroy_queue(hosts)
+    when "scan"
+      each_host(hosts, task_name) do |host|
+        if host.respond_to?(:scan)
+          add_flash(_("\"%{task}\": not supported for %{hostname}") % {:hostname => host.name, :task => (task_name || task)}, :error)
+        else
+          host.send(task.to_sym, session[:userid]) # Scan needs userid
+        end
+      end
     else
-      Host.where(:id => hosts).order("lower(name)").each do |host|
-        id = host.id
-        host_name = host.name
-        begin
-          if host.respond_to?(task)  # Run the task
-            if task == "scan"
-              host.send(task.to_sym, session[:userid]) # Scan needs userid
-            else
-              host.send(task.to_sym)          # Run the task
-            end
-          end
-        rescue StandardError => bang
-          add_flash(_("%{model} \"%{name}\": Error during '%{task}': ") % {:model => ui_lookup(:model => "Host"), :name => host_name, :task => (display_name || task)} << bang.message, :error) # Push msg and error flag
+      each_host(hosts, task_name) do |host|
+        if host.respond_to?(task) && host.is_available?(task)
```

Comment:
I assume this is for the grid screen where you can select multiple hosts?  What happens when you are on the summary screen?  I would expect the button to not show when is_available? returns false.